eng/dockerfile-templates/runtime/Dockerfile

```diff
@@ -1,20 +1,55 @@
 {{
+    set isServer2025 to OS_VERSION_NUMBER = "ltsc2025" ^
+    set isServer2022 to OS_VERSION_NUMBER = "ltsc2022" ^
+    set isServer2019 to OS_VERSION_NUMBER = "ltsc2019" ^
+    set isServer2016 to OS_VERSION_NUMBER = "ltsc2016" ^
+    set isNet35 to PRODUCT_VERSION = "3.5" ^
+    set isNet462 to PRODUCT_VERSION = "4.6.2" ^
+    set isNet472 to PRODUCT_VERSION = "4.7.2" ^
+    set isNet48 to PRODUCT_VERSION = "4.8" ^
+    set isNet481 to PRODUCT_VERSION = "4.8.1" ^
+
+    set kbExists to VARIABLES[cat("kb|", OS_VERSION_NUMBER, "|", PRODUCT_VERSION)] != void ^
     set is48SecurityRelease to (VARIABLES[cat("4.8-is-security-release|", OS_VERSION_NUMBER)] = "true") ^
-    set apply35Patch to (VARIABLES[cat("kb|", OS_VERSION_NUMBER, "|", PRODUCT_VERSION)] != void && PRODUCT_VERSION = "3.5" && OS_VERSION_NUMBER != "ltsc2019") ^
-    set applyPatch to VARIABLES[cat("kb|", OS_VERSION_NUMBER, "|", PRODUCT_VERSION)] != void &&
-        !(
-            (OS_VERSION_NUMBER = "ltsc2019" && PRODUCT_VERSION = "4.7.2") ||
-            (
+
+    set apply35Patch to kbExists && isNet35
+        && !isServer2019
+        && !isServer2025 ^
+
+    set applyPatch to kbExists
+        && !(isServer2025 && isNet481)
+        && !(isServer2019 && isNet472)
+        && !(
+                (isServer2019 && isNet472) ||
                 (
-                    (PRODUCT_VERSION = "4.8" && OS_VERSION_NUMBER != "ltsc2019") ||
-                    (PRODUCT_VERSION = "3.5" && OS_VERSION_NUMBER != "ltsc2019")
-                ) &&
-                is48SecurityRelease
-            )
-        )
+                    (
+                        (isNet48 && !isServer2019) ||
+                        (isNet35 && !isServer2019)
+                    ) &&
+                    is48SecurityRelease
+                )
+            ) ^
+
+    set winver to when(
+        isServer2016 || isServer2019 || isServer2022,
+        "windows10.0",
+        "windows11.0") ^
+
+    _ set applyPatch to when(isServer2025, isServer2025, applyPatch) ^
+    _ set apply35Patch to when(isServer2025, !isServer2025, apply35Patch) ^
+
+    set repo to "mcr.microsoft.com/windows/servercore" ^
+    set tag to cat(OS_VERSION_NUMBER, "-amd64") ^
+
+    set netfx3CabFilename to when(isServer2025,
+        "microsoft-windows-netfx3-ondemand-package~31bf3856ad364e35~amd64~en-us~.cab"
+        "microsoft-windows-netfx3-ondemand-package~31bf3856ad364e35~amd64~~.cab") ^
+
+    set fromImage to cat(repo, ":", tag)
+
 }}# escape=`
 
-FROM mcr.microsoft.com/windows/servercore:{{OS_VERSION_NUMBER}}-amd64
+FROM {{fromImage}}
```

Comment:
What is the value in this change? It seems more readable as it was before.

Reply:
I changed it back - was temporary for accommodating insider images.